snmp/datadog_checks/snmp/snmp.py

```diff
@@ -258,17 +258,17 @@ def fetch_oids(self, config, oids, enforce_constraints):
                     error = message
                 self.warning(message)
 
-            # if we fail move onto next batch
-            first_oid += self.oid_batch_size
+            finally:
```

Comment:
I don't think I see the point of the finally? It's not like moving the first_oid is useful when an exception is raised?

Reply:
Right, I ended up refactoring this using a `batches` utility that's hopefully clearer.